backend/graph/api/episode-resolver.utils.go

```diff
@@ -180,28 +189,39 @@ func (r *episodeResolver) getNextEpisodes(ctx context.Context, episodeID string,
 		return nil, err
 	}
 
-	next := cursor.NextCursor()
-	if next == nil {
-		ids, err := r.getNextFromShowCollection(ctx, episodeID)
+	l := 1
+	if limit != nil {
+		l = *limit
+	}
+
+	keys := cursor.NextKeys(1)
+	if len(keys) < l {
+		var ids []int
+		ids, err = r.getNextEpisodeIDsFromShowCollection(ctx, episodeID)
 		if err != nil {
 			return nil, err
 		}
-		if len(ids) > 0 {
-			return ids, nil
+
+		keys = appendNextEpisodeIDs(keys, ids)
```

Comment:
looks like these get shuffled even when the cursor isnt in shuffle mode

Reply:
They should be shuffled.